MobilityData/mobility-feed-api
# feat: added `official` value in endpoint params and responses

**Summary:**
This PR introduces the capability to filter feeds by their official status within the API. The implementation includes updates to existing methods, modifications to test data, and new test cases to ensure proper functionality.

#### **Enhancements to Feeds Filtering:**
- **[`feeds_api_impl.py`](diffhunk://#diff-d23c43cffed814911da274f6cc71a93b23a09cb6748afea6454455ec499e5df7R86-R124):**
  - Added a new helper method `_get_latest_official_status_subquery` to facilitate filtering based on the latest official status.
  - Updated methods to include `is_official` as a filter parameter.

#### **Updates to Models and Filters:**
- **[`basic_feed_impl.py`](diffhunk://#diff-d787bcaa42ce3b6d9cdf52b7fae95e09835e3893ff31d082916a235425f3a892R26-R33):**
  - Modified the `from_orm` method to include the latest official status.
  - Added `joinedload` for `officialstatushistories` to optimize queries. [[1]](diffhunk://#diff-d787bcaa42ce3b6d9cdf52b7fae95e09835e3893ff31d082916a235425f3a892R26-R33) [[2]](diffhunk://#diff-d787bcaa42ce3b6d9cdf52b7fae95e09835e3893ff31d082916a235425f3a892L51-R60)
- **[`search_api_impl.py`](diffhunk://#diff-8ce840a02c54911c7ba9768be6f1640a8e083745ad492071db21e63cef69ea95L34-R34):**
  - Enhanced search queries to include `is_official` as a filter.
  - Improved functionality across multiple filter-related methods. [[1]](diffhunk://#diff-8ce840a02c54911c7ba9768be6f1640a8e083745ad492071db21e63cef69ea95L34-R34) [[2]](diffhunk://#diff-8ce840a02c54911c7ba9768be6f1640a8e083745ad492071db21e63cef69ea95R56-R81) [[3]](diffhunk://#diff-8ce840a02c54911c7ba9768be6f1640a8e083745ad492071db21e63cef69ea95L83-R93) [[4]](diffhunk://#diff-8ce840a02c54911c7ba9768be6f1640a8e083745ad492071db21e63cef69ea95R103-R114)

---

### **Testing Instructions:**
1. Create a tunnel to the development database.
2. Run the API locally.
3. Validate the filtering functionality by making requests to endpoints like:
   ```
   http://127.0.0.1:8080/v1/gtfs_feeds?official=true
   ```
   Ensure that the behavior matches expectations.

Please make sure these boxes are checked before submitting your pull request - thanks!
